src/config/syscheck-config.c

```diff
@@ -361,16 +369,19 @@ int read_reg(syscheck_config *syscheck, char *entries, int arch, char *tag)
 
         /* Add entries - look for the last available */
         i = 0;
+        bool duplicated_entry = false;
         while (syscheck->registry && syscheck->registry[i].entry) {
             /* Duplicated entry */
             if (syscheck->registry[i].arch == arch && strcmp(syscheck->registry[i].entry, tmp_entry) == 0) {
```

Comment:
I think the overwriting of records is not being done properly. In this loop we check that the record is already inserted, but in the `dump_syscheck_entry()` function we check against the directories not the records, please check this behavior.